### PR TITLE
wp4:Routplanning

### PR DESCRIPTION
**Pull request by PerKristianNet**
Dijkstra added with retracing steps to target
needs integration to database

### CODE REVIEW

File: TelstarLogistics/RouteplannerTests/DjikstraTests.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+            int[,] graph = new int[,]
+            {
+                { 0, 0, 2 }, //0 connections
+                { 0, 0, 2 }, //1 ----||----
+                { 2, 2, 0 }, //2 ----||----
+            };
+            int sourceNode = 0;
+            int destinationNode = 1;
+
+
```
Comment (KonradMagiera) [inline TelstarLogistics/RouteplannerTests/DjikstraTests.cs]: remove unused empty spaces

File: TelstarLogistics/RouteplannerTests/DjikstraTests.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+            int[] distances = TelstarLogistics.Services.RoutePlanning.Dijkstra.ComputePaths(graph, sourceNode, out var parents);
+            var path = TelstarLogistics.Services.RoutePlanning.Dijkstra.GetPath(destinationNode, parents);
+            Assert.IsTrue(path[0] == 0 && path[1] == 2 && path[2] == 1);
+        }
+    }
+
```
Comment (KonradMagiera) [inline TelstarLogistics/RouteplannerTests/DjikstraTests.cs]: empty white spaces

File: TelstarLogistics/TelstarLogistics/Services/RoutePlanning/Dijkstra.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+                    parent[j] = minDistanceNode;
+                }
+            }
+        }
+        return distance;
+    }
+
+
+
+    static int MinimumDistanceNode(int[] distance, bool[] visited)
```
Comment (KonradMagiera) [inline TelstarLogistics/TelstarLogistics/Services/RoutePlanning/Dijkstra.cs:36]: should this method be private maybe?

File: TelstarLogistics/TelstarLogistics/Services/RoutePlanning/Dijkstra.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                min = distance[i];
+                minIndex = i;
+            }
+        }
+        return minIndex;
+    }
+
+    public static List<int> GetPath(int endVertex, int[] parents)
+    {
+        var path = RetracePath(endVertex, parents);
```
Comment (TheaBroholt) [inline TelstarLogistics/TelstarLogistics/Services/RoutePlanning/Dijkstra.cs]: This doesn't follow C# best practice for "var"-usage

File: TelstarLogistics/RouteplannerTests/DjikstraTests.cs (hunk 2 above)
Comment (KonradMagiera) [reply TelstarLogistics/RouteplannerTests/DjikstraTests.cs]: general note: there are multiple empty lines in your PR if you could run through all files

File: TelstarLogistics/TelstarLogistics/Services/RoutePlanning/DijkstraGraph.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        var graph = new int[cityCount,cityCount];
+        //iterate all connections
+
+        return graph;
+
+    }
+
+    
+
+    private static string printSolution(int startVertex,
```
Comment (TheaBroholt) [inline TelstarLogistics/TelstarLogistics/Services/RoutePlanning/DijkstraGraph.cs]: Unused - can we delete?

File: TelstarLogistics/RouteplannerTests/DjikstraTests.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+                { 0, 0, 2 }, //1 ----||----
+                { 2, 2, 0 }, //2 ----||----
+            };
+            int sourceNode = 0;
+            int destinationNode = 1;
+
+
+
+            int[] distances = TelstarLogistics.Services.RoutePlanning.Dijkstra.ComputePaths(graph, sourceNode, out var parents);
+            var path = TelstarLogistics.Services.RoutePlanning.Dijkstra.GetPath(destinationNode, parents);
```
Comment (TheaBroholt) [inline TelstarLogistics/RouteplannerTests/DjikstraTests.cs]: var is clear?

**Review by TheaBroholt — APPROVED** [review]

**Pull request merged**